docs/cloud_build_standards.md

```diff
@@ -0,0 +1,90 @@
+
+Please familiarise yourself with
+
+- Tagging Concepts and what services support / don't support [tagging](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html)
+- [s3 naming restrictions](https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html)
+
+### Tagging considerations
+
+- Tags can be applied anytime
+- Cost allocation reports are only available from the point tags were activated
+- After creating a tag [key] for costs, it must be marked activated ? added as a cost allocation tag, otherwise it will not be visible in the Cost Explorer
+
+### General Tagging Standards
+
+Below are a list of tagging standards to consider. Not all will be applicable i.e. based on the above shared account concepts (it's hosted in an account with other services), if it's clustered or not.
+
+All of the below should apply however, to all resources where you can.
+
+
+| Type | Description | Usage | Optional |
+| ---- | ------ | ------ | ------|
+| `TYPE` | Service type | CI | No |
+| `ENV` | Type-ENV  | ci-dev, ops-prod, notprod-uat, notprod-ppd | No |
+| `CLUSTER` | Cluster Name | cluster1, cluster2 | Yes |
+| `NODE` | Node within the cluster | node1 | Yes |
+| `APPSTAGE` | Stage of the application and resources associated | Alpha, Beta, Live | Yes |
+| `ACCESS` | Private (pvt) or Public (pub) | pvt, pub | No |
+| `PROJECT SERVICE` | name of the service | projectx, projecty | No |
+| `PROJECT PORTFOLIO` | Name of the portfolio | portfoliox | yes |
+| `SERVICE TYPE` | Type of service | db, webapp, proxy | No |
+| `AZ` | availability zone number | 1a, 1b, 1c | No |
+| `COST-CODE` | cost code if a shared account | 1234-5678 | yes |
+| `REPO` | Repository related to the creation | repo1 | yes |
+| `OWNER/TEAM` | Name of the team if a shared account | team1 | yes |
+
+### Additional Tagging Naming Standards
+
+| Type | Description | Usage | Optional |
+| ---- | ------ | ------ | ------|
+| `RDSTYPE` | database type | Postgres, mysql, oracle | RDSTYPE: Oracle | No |
+
+### Automation and Cloud Rules
+
+- There should be **NO** manual changes to any resources everything should be done via code
+- Console access should be locked down to readonly to stop manual changes or changes being made that are not tracked via code
+- All code should be triggered via a CI job and not run manually
+- All code should be tested, linted, and planned
+- Terraform is the preferred choice of technology
+- Building AMI's is preferred over running configuration management tools post build phase [please read for sharing encrypted AMI's between accounts](https://aws.amazon.com/blogs/security/how-to-create-a-custom-ami-with-encrypted-amazon-ebs-snapshots-and-share-it-with-other-accounts-and-regions/)
+- Cloud config is preferred where CoreOS is used to bootstrap the OS
+- Cloud-init is preferred for non coreOS in exchange for backing AMI's, though building AMI's and baking images is preferred
```

Comment:
ba~~c~~king?

Reply:
I'll fix typo. Do you mean allow egress generically outbound and limit only on incoming?  or?

Reply:
yes, the former - unless there is a different direction?